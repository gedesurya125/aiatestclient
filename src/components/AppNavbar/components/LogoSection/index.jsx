import React from 'react'
import {styled} from '@mui/material/styles';
import logo from '../../../../assets/images/logoWhite.svg'

const LogoSectionContainer = styled('div')(
  ({theme}) => ({
    width: '200px',
    '& img':{
      width: '100%'
    },
    display: 'flex',
    alignItems: 'center'
  })
)


const LogoSection = () => {
  return (
    <LogoSectionContainer data-testid="logo-section">
      <img data-testid="image" src={logo} alt="..."/>
    </LogoSectionContainer>
  )
}

export default LogoSection
