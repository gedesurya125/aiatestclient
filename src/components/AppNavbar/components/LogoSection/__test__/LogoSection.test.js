import React from 'react';
import reactDOM from 'react-dom';
import {render, cleanup, fireEvent} from '@testing-library/react';
import renderer from 'react-test-renderer';
import LogoSection from '../index';

afterEach(cleanup);

it('render without crashing', () => {
  const div = document.createElement('div');
  reactDOM.render(<LogoSection/>, div);
  reactDOM.unmountComponentAtNode(div);
})

it('render component properly', () => {
  const {getByTestId} = render(<LogoSection/>);
  const container = getByTestId('logo-section');
  const imageTag = getByTestId('image');
  expect(container).toContainElement(imageTag);
  expect(imageTag).toHaveAttribute('src', expect.stringContaining('logo'))
})

it('Matcher snapshot 1', () => {
  const tree = renderer.create(<LogoSection/>).toJSON();
  expect(tree).toMatchSnapshot();
})

