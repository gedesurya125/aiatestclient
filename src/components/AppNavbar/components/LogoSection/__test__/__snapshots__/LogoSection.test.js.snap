// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matcher snapshot 1 1`] = `
<div
  className="css-1yd8edg"
  data-testid="logo-section"
>
  <img
    alt="..."
    data-testid="image"
    src="logoWhite.svg"
  />
</div>
`;
